tests/components/bluesound/test_media_player.py

```diff
@@ -0,0 +1,348 @@
+    await hass.services.async_call(
+        "media_player",
+        "volume_set",
+        {"entity_id": "media_player.player_name1111", "volume_level": 0.5},
+        blocking=True,
+    )
+
+    player_mocks.player_data.player.volume.assert_called_once_with(level=50)
+
+
+async def test_volume_mute(
+    hass: HomeAssistant, setup_config_entry: None, player_mocks: PlayerMocks
+) -> None:
+    """Test the media player volume mute."""
+    await hass.services.async_call(
+        "media_player",
+        "volume_mute",
+        {"entity_id": "media_player.player_name1111", "is_volume_muted": True},
+        blocking=True,
+    )
+
+    player_mocks.player_data.player.volume.assert_called_once_with(mute=True)
+
+
+async def test_volume_up(
+    hass: HomeAssistant, setup_config_entry: None, player_mocks: PlayerMocks
+) -> None:
+    """Test the media player volume up."""
+    await hass.services.async_call(
+        "media_player",
+        "volume_up",
+        {"entity_id": "media_player.player_name1111"},
+        blocking=True,
+    )
+
+    player_mocks.player_data.player.volume.assert_called_once_with(level=11)
+
+
+async def test_volume_down(
+    hass: HomeAssistant, setup_config_entry: None, player_mocks: PlayerMocks
+) -> None:
+    """Test the media player volume down."""
+    await hass.services.async_call(
+        "media_player",
+        "volume_down",
+        {"entity_id": "media_player.player_name1111"},
+        blocking=True,
+    )
+
+    player_mocks.player_data.player.volume.assert_called_once_with(level=9)
+
+
+async def test_attributes_set(
+    hass: HomeAssistant, setup_config_entry: None, player_mocks: PlayerMocks
+) -> None:
+    """Test the media player attributes set."""
+    state = hass.states.get("media_player.player_name1111")
+    assert state.state == "playing"
+    assert state.attributes["volume_level"] == 0.1
+    assert state.attributes["is_volume_muted"] is False
+    assert state.attributes["media_content_type"] == "music"
+    assert state.attributes["media_position"] == 2
+    assert state.attributes["shuffle"] is False
+    assert state.attributes["master"] is False
+    assert state.attributes["friendly_name"] == "player-name1111"
+    assert state.attributes["media_title"] == "song"
+    assert state.attributes["media_artist"] == "artist"
+    assert state.attributes["media_album_name"] == "album"
+
+
+async def test_status_updated(
+    hass: HomeAssistant,
+    setup_config_entry: None,
+    player_mocks: PlayerMocks,
+) -> None:
+    """Test the media player status updated."""
+    pre_state = hass.states.get("media_player.player_name1111")
+    assert pre_state.state == "playing"
+    assert pre_state.attributes["volume_level"] == 0.1
+
+    status = player_mocks.player_data.status_store.get()
+    status = dataclasses.replace(status, state="pause", volume=50, etag="changed")
+    player_mocks.player_data.status_store.set(status)
+
+    await asyncio.sleep(0)
```

Comment:
Out of curiosity, does `await hass.block_until_done()` fix this? (or maybe with `wait_background_tasks=True`)

Reply:
`wait_background_tasks=True` will block forever, because the long-polling loop is running forever(as long as the config_entry is loaded). 

`await hass.block_until_done()` works the same way `await asyncio.sleep()` works: It yields to the event loop and the changed state can be processed.

Either way is kind of ugly, but i don't see any way to wait for the change state to be processed inside of the long-polling loop.

Reply:
We rather do the async block till done, we do that everywhere and that's less confusing :)